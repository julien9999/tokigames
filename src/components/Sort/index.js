import React from 'react';
import { setFilter } from '../../actions/filterActions';
import { SHOW_ALL_FILTER, SHOW_CHEAP, SHOW_BUSINESS } from '../../constants/actionTypes';
import FilterLink from '../FilterLink'

export default function Sort() {
  return (
    <div className='container'>
        <div className='row'>
          <div className='col-12'> 
            <h5 class="card-title margin-top-twenty">Sort options:</h5>
          </div>
          <div className='col-12'>       
            <ul class="nav nav-pills nav-fill">
              <FilterLink action={setFilter} type={SHOW_ALL_FILTER} property="filter">All</FilterLink>
              <FilterLink action={setFilter} type={SHOW_CHEAP} property="filter">Cheap</FilterLink>
              <FilterLink action={setFilter} type={SHOW_BUSINESS} property="filter">Business</FilterLink>
            </ul>
          </div>
        </div>
    </div>
  );
}

