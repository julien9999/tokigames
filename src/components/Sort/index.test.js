import React from 'react';
import renderer from 'react-test-renderer';
import Sort from './index';
import { Provider } from 'react-redux';
import configure from '../../store/configureStore';

const store = configure();

describe('<Sort>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
            <Sort />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

