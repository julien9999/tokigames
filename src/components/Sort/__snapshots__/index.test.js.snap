// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <h5
        class="card-title margin-top-twenty"
      >
        Sort options:
      </h5>
    </div>
    <div
      className="col-12"
    >
      <ul
        class="nav nav-pills nav-fill"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            onClick={[Function]}
          >
            All
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link null"
            onClick={[Function]}
          >
            Cheap
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link null"
            onClick={[Function]}
          >
            Business
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
